consensus/state.go

```diff
@@ -4,53 +4,45 @@ import (
 	"sync"
 )
 
-// The zero address and the zero currency are convenience variables.
+// The ZeroUnlockHash and ZeroCurrency are convenience variables.
 var (
 	ZeroUnlockHash = UnlockHash{0}
 	ZeroCurrency   = NewCurrency64(0)
 )
 
 // The State is the object responsible for tracking the current status of the
-// blockchain. It accepts blocks and maintains an understanding of competing
-// forks. The State object is responsible for maintaining consensus.
+// blockchain. Broadly speaking, it is responsible for maintaining consensus.
+// It accepts blocks and constructs a blockchain, forking when necessary.
 type State struct {
-	// The blockRoot is the block node that contains the genesis block, which
-	// is the foundation for all other blocks. blockNodes form a tree, each
-	// having many children and pointing back to the parent.
+	// The blockRoot is the block node that contains the genesis block.
 	blockRoot *blockNode
 
-	// badBlocks and blockMap keep track of known blocks. badBlocks keeps track
-	// of invalid blocks and is used exclusively for DoS prevention. blockMap
-	// points only to blocks that exist in some competing fork within the
-	// blockchain.
-	badBlocks map[BlockID]struct{}
+	// blockMap and badBlocks keep track of seen blocks. blockMap holds all
+	// valid blocks, including those not on the main blockchain. badBlocks
+	// is a "blacklist" of blocks known to be invalid.
 	blockMap  map[BlockID]*blockNode
+	badBlocks map[BlockID]struct{}
 
 	// currentPath and currentBlockID track which blocks are currently accepted
 	// as the longest known blockchain.
-	currentBlockID BlockID
 	currentPath    map[BlockHeight]BlockID
+	currentBlockID BlockID
```

Comment:
currentPath and currentBlockID are really parts of the consensus set.

Though, on a super pedantic level some people might disagree with me.